### PR TITLE
feat(sql): log the size of executions when they complete

### PR DESCRIPTION
**Pull request by dbyron-sf**
to make changes in size over time more observable.  This makes it easier to see the impact of features like the artifact store.

Not setting the size in RedisExecutionRepository because the extra objectMapper.writeValueAsString is potentially expensive and I'm not sure how many folks are using redis for this.

### CODE REVIEW

**Comment by dbyron-sf** [comment]
@mergifyio update

**Comment by jasonmcintosh** [comment]
I'd wonder if this would be better as a metric vs. part of artifact storage?  Not saying "no" as it'd be interesting to track it here...

**Comment by dbyron-sf** [comment]
> I'd wonder if this would be better as a metric vs. part of artifact storage? Not saying "no" as it'd be interesting to track it here...

With execution id as one of the tags I think cardinality would be too high.  Probably config id too.  But I could totally see taking those out and making this a metric.

**Comment by jasonmcintosh** [comment]
Yeah :) I LIKE high cardinatlity metric data (YAY Honeycomb!) but... I KINDA lean towards removing them & keeping the data as a metric - would have LOVED pipeline size as a metric LONG ago to show exactly how bad this could be AND before/after artifact store support kinda stuff.  Not AGAINST it in the DB but I could see this being a VERY compelling metric to track.

**Comment by jasonmcintosh** [comment]
Application of course as a label, pipeline name as a label too ideally though that's iffy as that's getting high cardinality route.  BUT makes it a LOT easier to track activity by "bad pipeline behavior" AND find pipelines that are heavy hitters.

File: orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/CompleteExecutionHandler.kt [hunk 1]
```diff
@@ -64,6 +65,15 @@ class CompleteExecutionHandler(
         message.determineFinalStatus(execution) { status ->
           execution.updateStatus(status)
           repository.updateStatus(execution)
+          val executionContextSize = execution.getTotalSize()
+          if (executionContextSize.isPresent) {
+            log.info("completed pipeline execution size: {},{},{},{},{}",
```
[the remark this answers is not in the document]
Comment (jasonmcintosh) [reply orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/CompleteExecutionHandler.kt:70]: I'd absolutely go for the log message!

**Comment by dbyron-sf** [comment]
Maybe this is obvious, but the size of execution contexts can change a ton during execution.  Like, the completed size could be a small fraction of the max size.  Reducing the max size can be a big win even if the completed size stays the same.  Measuring this in some more thorough way is, from what I can tell, a non-trivial exercise.

**Comment by jasonmcintosh** [comment]
ORIGINAL thought was some metrics on the storage interfaces to track what was written to the storage layer... but haven't dug AS much on this

**Comment by dbyron-sf** [comment]
@jasonmcintosh is this OK to merge?  Not sure @xibz has time to give it a look.

**Comment by dbyron-sf** [comment]
@mergifyio update

**Comment by dbyron-sf** [comment]
@mergifyio update

**Review by jasonmcintosh — APPROVED** [review]

**Pull request merged**